plugins/http/guest-js/index.ts

```diff
@@ -30,519 +30,138 @@ declare global {
   }
 }
 
-/**
- * @since 2.0.0
- */
-interface Duration {
-  secs: number;
-  nanos: number;
-}
-
-/**
- * @since 2.0.0
- */
-interface ClientOptions {
-  /**
-   * Defines the maximum number of redirects the client should follow.
-   * If set to 0, no redirects will be followed.
-   */
-  maxRedirections?: number;
-  connectTimeout?: number | Duration;
-}
-
-/**
- * @since 2.0.0
- */
-enum ResponseType {
-  JSON = 1,
-  Text = 2,
-  Binary = 3,
-}
-
-/**
- * @since 2.0.0
- */
-interface FilePart<T> {
-  file: string | T;
-  mime?: string;
-  fileName?: string;
+async function readBody<T>(rid: number, kind: "blob" | "text"): Promise<T> {
+  return await window.__TAURI_INVOKE__("plugin:http|fetch_read_body", {
+    rid,
+    kind,
+  });
 }
 
-type Part = string | Uint8Array | FilePart<Uint8Array>;
+class TauriResponse extends Response {
```

Comment:
Is there a reason why we need to extend and override `Response` methods? Was `new Response(dataAsBytes)` not enough?